frontend/src/scenes/funnels/FunnelBarGraph.tsx

```diff
@@ -0,0 +1,140 @@
+        element: barRef,
+    })
+
+    return (
+        <div className="funnel-bar-wrapper">
+            <div ref={barRef} className="funnel-bar" style={{ width: `${percentage}%` }}>
+                <div
+                    ref={labelRef}
+                    className={`funnel-bar-percentage ${labelPosition}`}
+                    title={name ? `Users who did ${name}` : undefined}
+                    aria-role="progressbar"
+                    aria-valuemin={0}
+                    aria-valuemax={100}
+                    aria-valuenow={percentage}
+                >
+                    {humanizeNumber(percentage, 2)}%
+                </div>
+            </div>
+        </div>
+    )
+}
+
+interface ValueInspectorButtonProps {
+    icon?: JSX.Element
+    onClick: (e?: React.SyntheticEvent) => any
+    children: React.ReactNode
+    disabled?: boolean
+}
+
+function ValueInspectorButton({ icon, onClick, children, disabled = false }: ValueInspectorButtonProps): JSX.Element {
+    return (
+        <Button type="link" icon={icon} onClick={onClick} className="funnel-inspect-button" disabled={disabled}>
+            <span className="funnel-inspect-label">{children}</span>
+        </Button>
+    )
+}
+
+export function FunnelBarGraph({ layout = 'horizontal', steps: stepsParam }: FunnelBarGraphProps): JSX.Element {
+    const steps = stepsParam.sort((a, b) => a.order - b.order)
+    const referenceStep = steps[0] // Compare values to first step, i.e. total
+
+    function calcPercentage(numerator: number, denominator: number): number {
+        return (numerator / denominator) * 100 || 0
+    }
+
+    function humanizeOrder(order: number): number {
+        return order + 1
+    }
+
+    return layout === 'horizontal' ? (
+        <div>
+            {steps.map((step, i) => (
+                <section key={step.order} className="funnel-step">
+                    <div className="funnel-series-container">
+                        <div className={`funnel-series-linebox ${steps[i - 1] ? 'before' : ''}`} />
+                        <SeriesGlyph style={{ backgroundColor: '#fff', zIndex: 2 }}>
+                            {humanizeOrder(step.order)}
+                        </SeriesGlyph>
+                        <div className={`funnel-series-linebox ${steps[i + 1] ? 'after' : ''}`} />
+                    </div>
+                    <header>
+                        <div className="funnel-step-title">
+                            <PropertyKeyInfo value={step.name} />
+                        </div>
+                        <div className="funnel-step-metadata">
+                            {step.average_time >= 0 + Number.EPSILON && step.order !== 0 ? (
+                                <ValueInspectorButton icon={<ClockCircleOutlined />} onClick={() => {}} disabled>
+                                    {pluralize(step.average_time, 'hour')}
+                                </ValueInspectorButton>
+                            ) : null}
```

Comment:
Even though I'm using postgres and can see the time in the old graph, nothing shows up here. Not sure why...

Reply:
Removed `step !== 0` condition— I think this feature will need some work from a UX perspective but for now the value will at least be shown.